aws-android-sdk-cognitoidentityprovider/src/main/java/com/amazonaws/mobileconnectors/cognitoidentityprovider/CognitoUserPool.java

```diff
@@ -268,6 +273,12 @@ public CognitoUserPool(Context context, String userPoolId, String clientId, Stri
     public CognitoUserPool(Context context, String userPoolId, String clientId, String clientSecret, ClientConfiguration clientConfiguration, Regions region, String pinpointAppId) {
         initialize(context);
         this.context = context;
+        if (userPoolId.isEmpty() || clientId.isEmpty()) {
+            throw new IllegalArgumentException("Both UserPoolId and ClientId are required. ");
```

Comment:
Nit: extra space at the end of the exception string (also in the exceptions thrown below)

Reply:
Good catch!